internal/install/discovery/manifest_validator.go

```diff
@@ -0,0 +1,61 @@
+package discovery
+
+import (
+	"fmt"
+
+	"github.com/newrelic/newrelic-cli/internal/install/types"
+)
+
+type ManifestValidator struct {
+	validators []Validator
+}
+
+type Validator interface {
+	Execute(m *types.DiscoveryManifest) string
+}
+
+var (
+	ErrorPrefix       = "Installation requirements error:"
+	ErrorPrefixFormat = ErrorPrefix + " %s"
+)
+
+// NewManifestValidator returns a new instance of ManifestValidator.
+func NewManifestValidator() *ManifestValidator {
+	mv := ManifestValidator{
+		validators: []Validator{},
+	}
+
+	mv.validators = append(mv.validators, NewOsValidator())
+	mv.validators = append(mv.validators, NewOsWindowsValidator())
+
+	return &mv
+}
+
+func (mv *ManifestValidator) Execute(m *types.DiscoveryManifest) string {
```

Comment:
I think @zlesnr said the same elsewhere, but feels like these errors should be propagated up and returned as error vs string.